Endefector

Meerlikeskeer opened the pull request:
preparing to merge code for robot subsystems.

Meerlikeskeer committed:
Simple Logging and Sim, js like akit b w/ doglog

Meerlikeskeer committed:
Swerve Rewrite for Doglog

Meerlikeskeer committed:
Moving Commands out of drive

Meerlikeskeer committed:
Adding Commands to Robot Container and Autos

Meerlikeskeer committed:
10 showed me smthn super cool

jaidenlikesrice committed:
Cooked up some autos

Taxi, One Note, Two Note, and Three Note Autos. Autos still need to be put into AutoRoutines and RobotContainer in the future.

Meerlikeskeer committed:
maybe this is move to point...??

Meerlikeskeer committed:
pushing again cus last time it didnt push lol

AndreTommySalehi committed:
made climb branch

Meerlikeskeer committed:
fixed sim odometry

Meerlikeskeer committed:
stupid fieldconstants pissing me off

jaidenlikesrice committed:
UpdatedTunerConstants

DoesNotWork-NeedsFixing

Meerlikeskeer committed:
fuck tuner constants, my constants all the way

litlizlovesprog committed:
Rollers & Wrist

Both are mostly done I just have to deal with all the DogLog part.

litlizlovesprog committed:
Finished endefector

The only thing thats missing is the pid values, I have no idea how to get them.

litlizlovesprog committed:
Deleted claw & wrote RobotContainer

Meerlikeskeer committed:
just another swerve rewrite

Meerlikeskeer committed:
lets hope this one works otherwise we are cooked.

Meerlikeskeer committed:
hey robot see stuff maybe idk

Meerlikeskeer committed:
fixed the swerev, jaiden will tune =-D

Meerlikeskeer committed:
pushing changes to temp branch

Meerlikeskeer committed:
btuh

AndreTommySalehi committed:
climb prolly done

it's not deploying but I'll fix later

Meerlikeskeer committed:
Merge branch 'Vision' into vision-and-swerve

AndreTommySalehi committed:
added some commands and changed motor IDs

Meerlikeskeer committed:
updated to 2.1 and commented to show andre how

Meerlikeskeer commented on 2025_NoName/src/main/java/frc/robot/endefector/rollers/RollersIO.java:
```diff
@@ -0,0 +1,27 @@
+// // import dev.doglog.DogLog;
+
+public abstract class RollersIO {
+  public static class RollersIOInputs {
```
showing andre how to comment

Meerlikeskeer committed:
Update RollersIOSim.java

AndreTommySalehi commented on 2025_NoName/src/main/java/frc/robot/util/SimLog.java:
```diff
@@ -0,0 +1,25 @@
+
+public class SimLog extends DogLog {
+  public static void log(String key, DCMotorSim motor) {
+    log(key + "RollersVelocity", motor.getAngularVelocityRPM() / 60);
+    log(key + "RollersVoltage", motor.getInputVoltage());
+    log(key + "RollersAmps", motor.getCurrentDrawAmps());
+    log(key + "RollersTemp", 60);
+    
+    log(key + "WristVelocity", motor.getAngularVelocityRPM() / 60);
+    log(key + "WristVoltage", motor.getInputVoltage());
+    log(key + "WristAmps", motor.getCurrentDrawAmps());
+    log(key + "WristTemp", 60);
+    log(key + "WristPosition", motor.getAngularPositionRotations());
+
+    log(key + "ClawVelocity", motor.getAngularVelocityRPM() / 60);
+    log(key + "ClawVoltage", motor.getInputVoltage());
+    log(key + "ClawAmps", motor.getCurrentDrawAmps());
+    log(key + "ClawTemp", 60);
+    log(key + "ClawPosition", motor.getAngularPositionRotations());
```
Only log the basic motor inputs once because this is for you to log the general inputs of every sim motor like the temp, amps, velocity, voltage, etc. But for your inputs that are more specific to each motor like wristPosition, log those in their own sim layer using doglog.log.

Code for Simlog:
log(key + "/Velocity", motor.getAngularVelocityRPM() / 60);
    log(key + "/Voltage", motor.getInputVoltage());
    log(key + "/Amps", motor.getCurrentDrawAmps());
    log(key + "/Temp", 60);

AndreTommySalehi commented on 2025_NoName/src/main/java/frc/robot/endefector/wrist/WristIOSim.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public void updateInputs() {
+        SimLog.log("WristVelocity", wristSim);
+        SimLog.log("WristVoltage", wristSim);
+        SimLog.log("WristAmps", wristSim);
+        SimLog.log("WristTemp", wristSim);
+        SimLog.log("WristPosition", wristSim);
```
You are trying to log every individual input that SimLog already logs for you, which results in you just logging those same inputs five times under different motor names. Instead, just write one SimLog line and give it the motor name and the sim motor you are using. After that add in the inputs that are specific to this wrist sim motor using Doglog.

Code for WristIOSim:
    @Override
    public void updateInputs() {
        SimLog.log("WristSimMotor", wristSim);

        DogLog.log("Wrist/TargetPosition", passedInPositon);
        DogLog.log("Wrist/CurrentPosition", currentPosition);
        DogLog.log("Wrist/Position", wristSim.getAngularPositionRotations()); // this input used to be in SimLog but it should be here instead because it is not general to every sim motor.

    }

AndreTommySalehi commented on 2025_NoName/src/main/java/frc/robot/util/MotorLog.java:
```diff
@@ -0,0 +1,49 @@
+
+        //Rollers
+        log(key + "Rollers", motor.getVelocity().getValueAsDouble());
+        
+        log(key + "Rollers", motor.getSupplyVoltage().getValueAsDouble());
+        
+        log(key + "Rollers", motor.getStatorCurrent().getValueAsDouble());
+
+        log(key + "Rollers", motor.getDeviceTemp().getValueAsDouble());
+
+
+        //Wrist velocityRadsPerSec
+        log(key + "Wrist", motor.getVelocity().getValueAsDouble());
+
+        //Wrist appliedVoltage
+        log(key + "Wrist", motor.getSupplyVoltage().getValueAsDouble());
+
+        //Wrist currentAmps
+        log(key + "Wrist", motor.getStatorCurrent().getValueAsDouble());
+
+        //Wrist tempCelcius
+        log(key + "Wrist", motor.getDeviceTemp().getValueAsDouble());
+
+        //Wrist setpointAngle
+        log(key + "Wrist", motor.getPosition().getValueAsDouble());
+
+
+        log(key + "Claw", motor.getVelocity().getValueAsDouble());
+
+        //Claw appliedVoltage
+        log(key + "Claw", motor.getSupplyVoltage().getValueAsDouble());
+
+        //Claw currentAmps
+        log(key + "Claw", motor.getStatorCurrent().getValueAsDouble());
+
+        //Claw tempCelcius
+        log(key + "Claw", motor.getDeviceTemp().getValueAsDouble());
+
+        //Claw setpointAngle
+        log(key + "Claw", motor.getPosition().getValueAsDouble());
```
This is the same issue as the SimLog. You don't need to log the general inputs for every motor, just log them once because MotorLog is for every motor. And for inputs that are specific to one motor like wrist position, log them using DogLog.log in the talonfx layer.

Code for MotorLog:

    public static void log(String key, TalonFX motor){

        //Rollers
        log(key + "/Velocity", motor.getVelocity().getValueAsDouble());
        
        log(key + "/Voltage", motor.getSupplyVoltage().getValueAsDouble());
        
        log(key + "/CurrentAmps", motor.getStatorCurrent().getValueAsDouble());

        log(key + "/Temp", motor.getDeviceTemp().getValueAsDouble());

    }

AndreTommySalehi commented on 2025_NoName/src/main/java/frc/robot/endefector/wrist/WristIOTalonFX.java:
```diff
@@ -0,0 +1,94 @@
+    public void updateInputs() {
+      MotorLog.log("Wrist", wristMotor);
+      DogLog.log("Wrist/TargetPosition", passedInPosition);
+      DogLog.log("Wrist/CurrentPosition", currentPosition);
```
Everything here is fine just log the claw position here using DogLog.log since that only applies to this motor.

Code for WristIOTalonFX:

    @Override
    public void updateInputs() {
      MotorLog.log("Wrist", wristMotor);

      DogLog.log("Wrist/TargetPosition", passedInPosition);
      DogLog.log("Wrist/CurrentPosition", currentPosition);
      DogLog.log("Wrist/Position", wristMotor.getPosition().getValueAsDouble());
  }

litlizlovesprog committed:
Fixed my logging issues

litlizlovesprog committed:
Merge branch 'Endefector' of https://github.com/Robodox-599/2025_NoName into Endefector

litlizlovesprog committed:
Added a CANcoder to wrist

Meerlikeskeer committed:
fixed swerve and added vision camera to swerve

Meerlikeskeer committed:
modifying vision to change system for single tags

Meerlikeskeer committed:
Revert "modifying vision to change system for single tags"

This reverts commit 430b43d947fa7a71003384664ed11a171e9b6fbe.

Meerlikeskeer committed:
vision removal from swerve branch.

this is the swerve only branch, no merging in any vision. there is the other vision and swerve branch for that.

jaidenlikesrice committed:
Tuned Swerve and fixed stuff

Commented out any vision stuff

Meerlikeskeer committed:
k1 changes, swerve works

jaidenlikesrice committed:
Created Auto Paths

Created auto path sections for one side. Need to create more for the other side and implement the branching autos into code.

jaidenlikesrice committed:
Merge branch 'swerve' of https://github.com/Robodox-599/2025_Diatomologist into swerve

Meerlikeskeer committed:
slight debugging and cleaning up

Meerlikeskeer committed:
Merge branch 'swerve' of https://github.com/Robodox-599/2025_Diatomologist into swerve

jaidenlikesrice committed:
More Choreo Paths

Named Choreo paths to be more specific on the reef position.

jaidenlikesrice committed:
Added Choreo Variables

Will add more for each reef position and both HP stations (makes it easier to create autos and have consistency)

jaidenlikesrice committed:
Added offsets for reef

litlizlovesprog committed:
Added the setState method

jaidenlikesrice committed:
Fixed Swerve Odom

Swerve odom seems pretty good, but the joystick controls are inverted (need to fix)

Meerlikeskeer committed:
Made Progress on Auto Align/ Score + Clean up code

litlizlovesprog committed:
Fixed some issues

litlizlovesprog committed:
Fixed some more issues

for some reason my code decided to kill itself but I fixed it

AndreTommySalehi committed:
Merge pull request #5 from Robodox-599/Climb

Climb

AndreTommySalehi committed:
Merge branch 'main' into Endefector

AndreTommySalehi committed:
Merge pull request #13 from Robodox-599/testing-main

Merging LEDs into test main

litlizlovesprog committed:
Revert "Merging LEDs into test main"

litlizlovesprog committed:
Merge pull request #14 from Robodox-599/revert-13-testing-main

Revert "Merging LEDs into test main"

AndreTommySalehi committed:
Merge pull request #15 from Robodox-599/LEDs

LEDs

M-i-c-h-a-e-l-599 committed:
Change the name of rollers and wirst

Changed from Wrist.java to intakeWrist.java. And from rollers.java. To intakeRollers

AndreTommySalehi committed:
Merge pull request #17 from Robodox-599/Algea-Ground-Intake

Algae ground intake

litlizlovesprog committed:
Revert "Algae ground intake"

litlizlovesprog committed:
Merge pull request #18 from Robodox-599/revert-17-Algea-Ground-Intake

Revert "Algae ground intake"

litlizlovesprog committed:
Revert "LEDs"

litlizlovesprog committed:
Merge pull request #19 from Robodox-599/revert-15-LEDs

Revert "LEDs"

litlizlovesprog committed:
Merge branch 'testing-main' into Endefector

litlizlovesprog committed:
Merged Endefector

jaidenlikesrice committed:
Fixed Swerve Bugs and Tested Autos

Fixed some bugs in the swerve and I tested the taxi auto. It currently does not work, I need to tune the PID and fix the issue of it turning without it being commanded to.

jaidenlikesrice committed:
Tried Tuning Sim PID

I was unable to do so, but I think the issue lies in incorrect units.

litlizlovesprog committed:
Added CANrange

litlizlovesprog committed:
Merge branch 'auto-align' into Endefector

jaidenlikesrice committed:
testing endefector
